### PR TITLE
Packaging for 1.0

### PR DESCRIPTION
**Pull request by blocknotes**
- README improvements
- Usage examples for processors and importers
- Update Gemspec and extension itself via` solidus_dev_support`
- Whatever is needed for a **1.0** release

### CODE REVIEW

File: examples/processors/notify_on_failure.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+module SolidusImporter
+  module Processors
+    ##
+    # This processor sends an email if the process of a single row is failing.
+    #
+    # Install: add the class to processors in solidus_importer configuration
+    class NotifyOnFailure < Base
+      def call(context)
+        return if context[:success]
```
Comment (elia) [inline examples/processors/notify_on_failure.rb:11]: shouldn't this return the original context here?

File: README.md [hunk 2]
```diff
@@ -31,6 +31,61 @@ SolidusImporter::ProcessImport.new(import).process
...
+    SolidusImporter::Processors::Log
+  ]
+  config.importer.order_processors = %w[
+    SolidusImporter::Processors::Order
+    SolidusImporter::Processors::Log
+  ]
+  config.importer.customer_processors = %w[
+    SolidusImporter::Processors::Customer
+    SolidusImporter::Processors::Log
+  ]
```
Comment (elia) [inline README.md:59]: This configuration is not what we ended up with, we should either amend the current configuration class to separate the config for each processor and/or update this section of the readme.

File: README.md [hunk 3]
```diff
@@ -31,6 +31,61 @@ SolidusImporter::ProcessImport.new(import).process
...
+CustomLoggerProcessor = ->(context) {
+  context.merge(logger: CUSTOM_LOGGER)
+}
+# Replace the original Log processor with CustomLoggerProcessor
+Solidus.configuration.importer.customer_processors.map! do |processor|
+  if processor == 'SolidusImporter::Processors::Log'
+    'CustomLoggerProcessor'
+  else
+    processors
+  end
```
Comment (elia) [inline README.md:78]: as above, this code wouldn't work with the current implementation

File: README.md [hunk 4]
```diff
@@ -23,14 +23,9 @@ bundle exec rails g solidus_importer:install
 Sample code to import some products:
 
 ```ruby
-import = SolidusImporter::Import.new(import_type: 'products')
-import.import_file = 'some_path/sample_products.csv'
-import.save
-SolidusImporter::ProcessImport.new(import).process
+SolidusImporter.import! 'some_path/sample_products.csv', type: :products
```
Comment (elia) [inline README.md:26]: ❤️

File: examples/processors/notify_on_failure.rb (hunk 1 above)
Comment (MinasMazar) [reply examples/processors/notify_on_failure.rb:11]: No, actually processors only updates received context.

File: README.md (hunk 2 above)
Comment (MinasMazar) [reply README.md:69]: I've updated the README to fit the current implementation. The configurations improvements will come in a separate PR.

File: solidus_importer.gemspec [hunk 5]
```diff
@@ -7,17 +7,16 @@ Gem::Specification.new do |s|
...
   s.summary     = 'Solidus Importer'
-  s.description = 'Solidus Importer'
+  s.description = 'Solidus Importer extension to migrate data from other eCommerce systems'
   s.license     = 'BSD-3-Clause'
 
   s.author      = 'Nebulab SRL'
-  s.homepage    = 'https://www.nebulab.it'
-  # s.email       = 'TODO'
+  s.email       = 'contact@solidus.io'
+  s.homepage    = 'https://solidus.io'
```
Comment (elia) [inline solidus_importer.gemspec]: I think the home page should point to the repo's readme, not blocking though, we can deal with that after the merge

**Pull request merged**